packages/aws-serverless/scripts/buildLambdaLayer.ts

```diff
@@ -61,3 +64,113 @@ function fsForceMkdirSync(path: string): void {
   fs.rmSync(path, { recursive: true, force: true });
   fs.mkdirSync(path);
 }
+
+async function pruneNodeModules(): Promise<void> {
+  const entrypoints = [
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/esm/index.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/cjs/index.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/cjs/awslambda-auto.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/esm/awslambda-auto.js',
+  ];
+
+  const { fileList } = await nodeFileTrace(entrypoints);
+
+  const allFiles = getAllFiles('./build/aws/dist-serverless/nodejs/node_modules');
+
+  const filesToDelete = allFiles.filter(file => !fileList.has(file));
+  console.log(`Removing ${filesToDelete.length} unused files from node_modules.`);
+
+  for (const file of filesToDelete) {
+    try {
+      fs.unlinkSync(file);
+    } catch {
+      console.error(`Error deleting ${file}`);
+    }
+  }
+
+  console.log('Cleaning up empty directories.');
+
+  removeEmptyDirs('./build/aws/dist-serverless/nodejs/node_modules');
+
+  await minifyJavaScriptFiles(fileList);
```

Comment:
I'm not sure how good of an idea it is to do this manually here... 🤔 could we instead leverage a regular rollup build for this step, somehow?

Reply:
to be clear, I am not saying we def. need to do that, just that we should think about it :D what about other things our rollup build does, other than minifying, ...?